### ISSUE TITLE
Plugin breaks site if structure/sections deleted

### ISSUE THREAD

**Issue by cole007** [mmikkel/ChildMe-Craft#4]
Because Craft soft-deletes Sections and Entry Types if a section is deleted then the record for it technically still exists. This causes the plugin to kill the site as it is looking for a section against an entry type that is returning null.

See line 103 of `vendor/mmikkel/child-me/src/ChildMe.php`

Ideally we need to check for a section first or in the EntryTypeRecord lookup (line 101) we need to exclude returning results where dateDeleted is not null.

Have solved for time being with following quick/hotfix:

```
    $entryTypes = EntryTypeRecord::find()->all();
    foreach ($entryTypes as $entryType) {
        if($entryType->section) {
            $section = $entryType->section->handle;
            if (!isset($data['entryTypes'][$section])) {
                $data['entryTypes'][$section] = [];
            }
            $data['entryTypes'][$section][$entryType->id] = Craft::t('site', $entryType->name);
        }
    }
```

Will see if can sort out a pull request

**Comment by mmikkel** [comment]
Ah, good catch @cole007, thanks. A PR would be much appreciated, if not I'll add a fix myself ASAP :)

**Comment by mmikkel** [comment]
Thanks a lot for the PR, @cole007! 

I've done a little bit of testing now, and I'm actually unable to actually reproduce the original issue.

So when you delete a Section, both the section itself and all its entry types are flagged as "soft deleted" – i.e. their `deleteDated` column will have datetime value (any entry types deleted individually will also be soft deleted).

The curious thing is that I'm not seeing any soft deleted entry types returned by `EntryTypeRecord::find()->all();` (and hence, no errors thrown). The reason is that the [`craft\records\EntryType`](https://github.com/craftcms/cms/blob/develop/src/records/EntryType.php) class uses the [`SoftDeleteTrait`](https://github.com/craftcms/cms/blob/develop/src/db/SoftDeleteTrait.php) trait, which specifically adds a `'dateDeleted' => null` query param to criteria returned by the `find()` method:

```php
public static function find()
    {
        $query = parent::find();
        // todo: remove schema version condition after next beakpoint
        $schemaVersion = Craft::$app->getInstalledSchemaVersion();
        if (version_compare($schemaVersion, '3.1.19', '>=')) {
            $query->where(['dateDeleted' => null]);
        }
        return $query;
    }
```

So, I'm curious why you're seeing different behaviour on your end. Which Craft version are you on? Which error do you get, specifically (possible to share a stack trace?).

**Comment by mmikkel** [comment]
@cole007 I just cut a new release [1.0.5](https://github.com/mmikkel/ChildMe-Craft/releases/tag/1.0.5) to fix #2. The fix for that impacted the same piece of code that's relevant to this issue – can you test 1.0.5 and let me know if it also fixes your issue related to soft deleted Sections (I still can't reproduce)? Thanks!
